lib/rule-context.js

```diff
@@ -70,14 +70,18 @@ var PASSTHROUGHS = [
  * @param {array} options The configuration information to be added to the rule.
  * @param {object} settings The configuration settings passed from the config file.
  * @param {object} ecmaFeatures The ecmaFeatures settings passed from the config file.
+ * @param {string} parser The parser setting passed from the config file.
  */
-function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures) {
+function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures, parser) {
     // public.
     this.id = ruleId;
     this.options = options;
     this.settings = settings;
     this.ecmaFeatures = ecmaFeatures;
 
+    this.parser = require(parser);
+    this.parserName = parser;
```

Comment:
I think that probably warrants a `loadParser` function that is responsible for answering the question: relative to what path? CWD? `eslint` module? base of provided globs?

Incoming additional commit moves the `require` back out to `lib/eslint` for consistency with `parse`.